File: demo/ScoringSystem/ScoringSystem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ScoringSystem
{
    public class ScoringSystem
    {
        public int[] PlayerScores
        {
            get;
            private set;
        }

        public ScoringSystem(int numOfPlayers)
        {
            if(numOfPlayers < 1 || numOfPlayers > 4)
            {
                throw new Exception("Bad number of player input");
            }
            PlayerScores = new int[numOfPlayers];
            foreach(int player in PlayerScores)
            {
                PlayerScores[player] = 0;
            }
        }

        public void AddRoundScore(int[] PlayerRoundScores)
        {
            if(PlayerRoundScores == null || PlayerRoundScores.Length != PlayerScores.Length)
            {
                throw new Exception("Bad player round score inpute");
            }

            foreach(int player in PlayerScores)
            {
                PlayerScores[player] += PlayerRoundScores[player];
            }
        }


    }
}

File: demo/ScoreSystem/ScoringSystem.ts
class ScoringSystem {
    playerScores: number[];

    constructor(numOfPlayers: number) {
        if (numOfPlayers < 1 || numOfPlayers > 4) {
            throw new Error("Bad number of player input");
        }

        for (let i = 0; i < numOfPlayers; i++)
        {
            this.playerScores[i] = 0;
        }
    }

    AddRoundScore(playerRoundScores : number[]) {
        if (playerRoundScores == null || playerRoundScores.length != this.playerScores.length) {
            throw new Error("Bad player round score inpute");
        }

        for (let player of playerRoundScores) {
            this.playerScores[player] += playerRoundScores[player];
        }
    }

}